15_06_2017/non_uniform_disp_solver.pro

```diff
@@ -0,0 +1,552 @@
+Kb=1.380658d-23 ;m2 kg s-2 K-1, Boltzman constant
+mp=1.67262178d-27; kg, proton mass
+mu0=4.0d-7*!DPI ;[H·m−1] or [N·A−2]
+gama=5.d/3.d
+
+;internal
+a=4.0d*Mega;; 4.0d*Mega; loop radius  (Mm)
+;Lz=100.d*Mega; loop length (Mm)
+Lz=25.d*a; loop length (Mm)
+Ltr=1.d*a ;; Length of transition region (Mm)
+Te0=0.8*Mega; internal plasma temperature [MK]
+ne0=1.5d15 ; m-3
+rho0=ne0*mp;internal plasma density [kg m-3]
+B0=0.0015d; internal magnetic field [Tesla]
+Cs0=sqrt(2*gama*kb/mp*Te0) ; m s-1
+Va0=B0/sqrt(mu0*rho0) ; m s-1
+Ct0=Cs0*Va0/sqrt(Va0^2+Cs0^2)  ; m s-1
+beta0=2*Cs0^2/(gama*Va0^2)
+p0=B0^2/(2*mu0)*beta0 ; internal gas pressure [Pa]
+
+;external
+delrho=5.d; rho0/rho1, density ratio
+delTe=1.5d ; Te0/Te1
+;; delTe=Te0/Te1=rho1/rho0*p0/p1=beta0/beta1*delB^2
+beta1=(delTe*delrho*(1+1/beta0)-1)^(-1) ;
+delB=sqrt((1+beta1)/(1+beta0)) ; B0/B1
+delbeta=beta0/beta1
+Te1=Te0/delTe ;; external plasma temperature [MK]
+B1=B0/delB  ; external magnetic field [Tesla]
+ne1=ne0/delrho
+rho1=ne1*mp ; external plasma density [kg m-3]
+p1=B1^2/(2*mu0)*beta1 ; external gas pressure [Pa]
+Cs1=sqrt(2*gama*kb/mp*Te1) ; external sound speed [m s-1]
+Va1=B1/sqrt(mu0*rho1)
+Ct1=Cs1*Va1/sqrt(Cs1^2+Va1^2)
+Ck=sqrt((rho0*Va0^2+rho1*Va1^2)/(rho0+rho1))
+
+;; wave mode setup
+;; cos(mi*theta)*cos(ni*pi*z/L)*cos(2*pi*t/P)
+;; mi: azimuthal mode number
+;; ni: longitudinal mode number
+;;
+;; solve mi=1, ni=1
+;; fundamental fast kink mode Va0<Vp<Ck;
+
+mi=1 ; cos (m*theta) azimuthal mode number
+ni=1 ; longitudinal mode number
+Kz=ni*!dpi/Lz ; m-1
+Ka=Kz*a; m-1*m
+nvp=1000000
+Vpmin=(Ck-Va0)*0.8+Va0
+Vp=Vpmin+(1+dindgen(nvp))*(Ck-Vpmin)/(nvp+1)
+k0a=Ka*sqrt(-(Cs0^2-Vp^2)*(Va0^2-Vp^2)/((Cs0^2+Va0^2)*(Ct0^2-Vp^2))) ; k0a^2>0 unit [ka]; note the minus sign
+K1a=Ka*sqrt((Cs1^2-Vp^2)*(Va1^2-Vp^2)/((Cs1^2+Va1^2)*(Ct1^2-Vp^2))) ; K1a^2>0 unit [ka]
+;;;Homogenous dispersion relation
+disp=delrho*((Va0/Va0)^2-(Vp/Va0)^2)*k1a*dbeselk_deriv(k1a,mi)/dbeselk(k1a,mi)$
+  -((Va1/Va0)^2-(Vp/Va0)^2)*k0a*dbeselj_deriv(k0a,mi)/dbeselj(k0a,mi)
+
+disp_min=min(abs(disp),sol)
+vp0 = vp(sol) ; phase speed w0/kz
+k0a=k0a[sol]
+k0=k0a/a ; kz internal
+k1a=k1a[sol]
+k1=k1a/a ; kz external
+vp0=vp0[0]; m/s
+w0=vp0*kz ; rad/s
+wc0=Cs0*kz; acoustic frequency
+wc1=Cs1*kz
+wa0=Va0*kz; alfven frequency
+wa1=Va1*kz
+wt0=Ct0*kz; cusp frequency
+wt1=Ct1*kz
+
+;To make unitless quantities 
+unit_length = a ;[m]
+Unit_density = rho1 ;[kg m-3]
+unit_velocity = va0 ;[m s-1]
+unit_B = B1 ;[Tesla]
+unit_Temp = Te0 ;[K]
+unit_time = unit_length/unit_velocity ; [s]
+unit_pressure = unit_velocity^2*Unit_density ;[pa] 
+
+Ltr_norm = ltr/unit_length
+rho1_norm = rho1/Unit_density
+rho0_norm = rho0/Unit_density
+Va0_norm = Va0/unit_velocity
+Va1_norm = Va1/unit_velocity
+a_norm = a/unit_length
+B0_norm = B0/unit_B 
+kz_norm = kz*unit_length
+w_norm = 1/unit_time;va0_norm*kz_norm
+   
+;rho_k_tot_list= DCOMPLEXARR(N, 1)
+
+wk = Ck*kz ; kink frequency
+
+i = dcomplex(0,1) ; gives complex number
```

Comment:
This is fine but be careful that you're not looping over `i` anywhere and rewriting this.